react/features/recording/components/Recording/RecordButton.web.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+import { connect } from 'react-redux';
+
+import { translate } from '../../../base/i18n';
+import { getLocalParticipant } from '../../../base/participants';
+
+import AbstractRecordButton, {
+    _mapStateToProps as _abstractMapStateToProps,
+    type Props as AbstractProps
+} from './AbstractRecordButton';
+
+type Props = AbstractProps & {
+
+    /**
+     * Tooltip for the button when it's disabled in a certain way.
+     */
+    _fileRecordingsDisabledTooltipKey: ?string
+}
+
+/**
+ * An implementation of a button for starting and stopping recording.
+ */
+class RecordButton extends AbstractRecordButton<Props> {
+    iconName = 'icon-recEnable';
+    toggledIconName = 'icon-recDisable';
+
+    /**
+     * Constructor of the component.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.tooltip = props._fileRecordingsDisabledTooltipKey;
+    }
+
+    /**
+     * Implements {@code Component}'s componentWillReceiveProps.
+     *
+     * @inheritdoc
+     */
+    componentWillReceiveProps(newProps: Props) {
+        this.tooltip = newProps._fileRecordingsDisabledTooltipKey;
+    }
+
+    /**
+     * Helper function to be implemented by subclasses, which must return a
+     * boolean value indicating if this button is disabled or not.
+     *
+     * @override
+     * @protected
+     * @returns {boolean}
+     */
+    _isDisabled() {
+        return Boolean(this.tooltip);
```

Comment:
This is very weird to read. The tooltip should not affect if a component is disabled or not.

Reply:
Well, i reused the logic from Toolbox, see code there. I could introduce a boolean for this, but then it will be redundant which i don't like

Reply:
I want to say redundancy might be okay in this case for clarity. Maybe the point here is the web logic needs changing. Maybe if the user has configured through interfaceConfig to show the button, but recording isn't enabled, then the button should display but as disabled.

Reply:
Agreed with @virtuacoplenny here. Let's show it, but show it disabled. That's a reasonable expectation I'd say.

Reply:
Well, that needs to be discussed with Evan, because I got the current behaviour from him (through Damyan, and he pointed me to this PR too, that uses this logic: https://github.com/jitsi/jitsi-meet/pull/3075

Reply:
We can, I just don't really understand why do we need 2 variables to decide on the state if we can do the same with one. That's sacrificing efficiency for code readability.

Anyhow, I do it just to make sure this PR gets through finally.